选择音频崩溃 [BUG]

**Describe the bug**
选择音频后就崩溃了，Android 10

**To Reproduce**
选择 音频确定后崩溃

**Expected behavior**

```
java.lang.Error: java.io.FileNotFoundException: java.io.FileNotFoundException: No item at content://media/external/video/media/28
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1173)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641)
        at java.lang.Thread.run(Thread.java:929)
     Caused by: java.io.FileNotFoundException: java.io.FileNotFoundException: No item at content://media/external/video/media/28
        at android.database.DatabaseUtils.readExceptionWithFileNotFoundExceptionFromParcel(DatabaseUtils.java:149)
        at android.content.ContentProviderProxy.openTypedAssetFile(ContentProviderNative.java:705)
        at android.content.ContentResolver.openTypedAssetFileDescriptor(ContentResolver.java:1698)
        at android.content.ContentResolver.openAssetFileDescriptor(ContentResolver.java:1514)
        at android.content.ContentResolver.openInputStream(ContentResolver.java:1198)
        at top.kikt.imagescanner.core.cache.AndroidQCache.getCacheFile(AndroidQCache.kt:47)
        at top.kikt.imagescanner.core.utils.AndroidQDBUtils.getFilePath(AndroidQDBUtils.kt:309)
        at top.kikt.imagescanner.core.PhotoManager.getFile(PhotoManager.kt:145)
        at top.kikt.imagescanner.core.PhotoManagerPlugin$onHandlePermissionResult$6.invoke(PhotoManagerPlugin.kt:225)
        at top.kikt.imagescanner.core.PhotoManagerPlugin$onHandlePermissionResult$6.invoke(PhotoManagerPlugin.kt:26)
        at top.kikt.imagescanner.core.PhotoManagerPlugin$sam$java_lang_Runnable$0.run(Unknown Source:2)
```

**Screenshots**
![image](https://user-images.githubusercontent.com/8764899/88475943-f4450800-cf66-11ea-832c-409b2c0f8217.png)


**Flutter version**
```
 Flutter version 1.17.5 at /Users/.../sdk/flutter
    • Framework revision 8af6b2f038 (4 weeks ago), 2020-06-30 12:53:55 -0700
    • Engine revision ee76268252
    • Dart version 2.8.4
```

**Smartphone (please complete the following information):**
 - Device: 华为 P40
 - OS:Emui 10.1.0
 - Version:Android 10

--- a/android/src/main/kotlin/top/kikt/imagescanner/core/cache/AndroidQCache.kt
+++ b/android/src/main/kotlin/top/kikt/imagescanner/core/cache/AndroidQCache.kt
@@ -14,7 +14,7 @@ import java.io.FileOutputStream
 
 @RequiresApi(Build.VERSION_CODES.Q)
 class AndroidQCache {
-  
+
   fun getCacheFile(context: Context, id: String, displayName: String, isOrigin: Boolean): File {
     val originString =
       if (isOrigin) {
@@ -25,25 +25,26 @@ class AndroidQCache {
     val name = "$id${originString}_${displayName}"
     return File(context.cacheDir, name)
   }
-  
-  fun getCacheFile(context: Context, assetId: String, extName: String, type: Int, isOrigin: Boolean): File {
+
+  fun getCacheFile(context: Context, assetId: String, extName: String, type: Int, isOrigin: Boolean): File? {
     val targetFile = getCacheFile(context, assetId, extName, isOrigin)
-    
+
     if (targetFile.exists()) {
       return targetFile
     }
-    
+
     val contentResolver = context.contentResolver
     var uri =
-      if (type == 1)
-        Uri.withAppendedPath(MediaStore.Images.Media.EXTERNAL_CONTENT_URI, assetId)
-      else
-        Uri.withAppendedPath(MediaStore.Video.Media.EXTERNAL_CONTENT_URI, assetId)
-    
+        when (type) {
+          1 -> Uri.withAppendedPath(MediaStore.Images.Media.EXTERNAL_CONTENT_URI, assetId)
+          2 -> Uri.withAppendedPath(MediaStore.Video.Media.EXTERNAL_CONTENT_URI, assetId)
+          3 -> Uri.withAppendedPath(MediaStore.Audio.Media.EXTERNAL_CONTENT_URI, assetId)
+          else -> return null
+        }
     if (isOrigin) {
       uri = MediaStore.setRequireOriginal(uri)
     }
-    
+
     val inputStream = contentResolver.openInputStream(uri)
     val outputStream = FileOutputStream(targetFile)
     outputStream.use {
@@ -51,19 +52,19 @@ class AndroidQCache {
     }
     return targetFile
   }
-  
+
   fun saveAssetCache(context: Context, asset: AssetEntity, byteArray: ByteArray, isOrigin: Boolean = false) {
     val file = getCacheFile(context, asset.id, asset.displayName, isOrigin)
     if (file.exists()) {
       LogUtils.info("${asset.id} , isOrigin: $isOrigin, cache file exists, ignore save")
       return
     }
-    
+
     if (file.parentFile?.exists() != true) {
       file.mkdirs()
     }
     file.writeBytes(byteArray)
-  
+
     LogUtils.info("${asset.id} , isOrigin: $isOrigin, cached")
   }
 }

--- a/android/src/main/kotlin/top/kikt/imagescanner/core/utils/AndroidQDBUtils.kt
+++ b/android/src/main/kotlin/top/kikt/imagescanner/core/utils/AndroidQDBUtils.kt
@@ -306,7 +306,7 @@ object AndroidQDBUtils : IDBUtils {
 
   override fun getFilePath(context: Context, id: String, origin: Boolean): String? {
     val assetEntity = getAssetEntity(context, id) ?: return null
-    val cacheFile = androidQCache.getCacheFile(context, id, assetEntity.displayName, assetEntity.type, origin)
+    val cacheFile = androidQCache.getCacheFile(context, id, assetEntity.displayName, assetEntity.type, origin) ?: return null
     return cacheFile.path
   }
 

